zebra-state/src/service.rs

```diff
@@ -1554,6 +1551,44 @@ impl Service<ReadRequest> for ReadStateService {
                 .map(|join_result| join_result.expect("panic in ReadRequest::BestChainBlockHash"))
                 .boxed()
             }
+
+            // Used by get_block_template RPC.
+            #[cfg(feature = "getblocktemplate-rpcs")]
+            ReadRequest::ChainInfo => {
+                let timer = CodeTimer::start();
+
+                let state = self.clone();
+                let latest_non_finalized_state = self.latest_non_finalized_state();
+
+                // # Performance
+                //
+                // Allow other async tasks to make progress while concurrently reading blocks from disk.
+                let span = Span::current();
+                tokio::task::spawn_blocking(move || {
+                    span.in_scope(move || {
+                        let get_block_template_info = read::tip(
+                            latest_non_finalized_state.best_chain(),
+                            &state.db,
+                        )
+                        .map(|tip| GetBlockTemplateChainInfo {
+                            tip,
```

Comment:
I'm not sure if you're going to fix this later, but I'm letting you know just in case:

The time in this tip is the `nTime` for the chain tip block, but it is being used for `min_time` in the RPC.

The `min_time` consensus rule uses `median-time-past`, which is calculated from the `nTime` fields of 11 recent blocks:
> For each block other than the genesis block , nTime MUST be strictly greater than the median-time-past of that block.

https://zips.z.cash/protocol/protocol.pdf#blockheader

So you'll want to use `difficulty_adjustment.median_time_past()` + 1 second for the `min_time`:
https://github.com/ZcashFoundation/zebra/blob/2e50ccc8f3d067c5aa3c8924c3ccedabc1af477b/zebra-state/src/service/check/difficulty.rs#L310-L323